src/main/java/seedu/address/model/person/Address.java

```diff
@@ -14,13 +12,10 @@ public class Address {
     public static final String MESSAGE_CONSTRAINTS = "Addresses can take any values, and it should not be blank";
 
     /*
-     * The first character of the address must not be a whitespace,
+     * The first character of the address must not be a whitespace (ie start with alphanumeric or #)
      * otherwise " " (a blank string) becomes a valid input.
      */
-    public static final String VALIDATION_REGEX = "[^\\s].*";
-    public static final String DOOR_NUMBER_REGEX = "#\\d{2,3}-\\d{2,4}";
-    public static final String POSTAL_CODE_REGEX = "S\\d{6}"; // postal code must be Sxxxxxx
-
+    public static final String VALIDATION_REGEX = "^[a-zA-Z0-9#].*";
```

Comment:
I feel like this is a little bit too flexible, can we adjust the original pattern to support addresses without a unit number while making sure it's close to the original design?

Reply:
In the original code, `POSTAL_CODE_REGEX` is not used so only `DOOR_NUMBER_REGEX` will be used to check valid address. 

After modifications, it now checks that the first character of the address is alphanumeric or `#` to take in all possible address formats. 

Is this ok?